tests/test_other.py

```diff
@@ -9230,23 +9230,34 @@ def test_emscripten_metadata(self):
     self.assertTextDataIdentical('hello, world!\n', ret)
 
   @parameterized({
-    'O2': (False, ['-O2']), # noqa
-    'O2_emit': (True, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE']), # noqa
-    'O2_js_emit': (True, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE', '-s', 'WASM=0']), # noqa
-    'O2_closure_emit': (True, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE', '--closure', '1']), # noqa
-    'O2_closure_js_emit': (True, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE', '--closure', '1', '-s', 'WASM=0']), # noqa
+    # -O0 and -O1 do not optimize out JS comments, and license headers are just
+    # JS comments (so even multiple copies can be emitted).
+    # -O2+ will remove all comments, but preserve the license if that was asked
+    # for, and only one will exist.
+    'O0': (True, True, ['-O0']), # noqa
+    'O1': (True, True, ['-O1']), # noqa
+    'O2': (False, False,['-O2']), # noqa
+    'O2_emit': (True, False, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE']), # noqa
+    'O2_js_emit': (True, False, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE', '-s', 'WASM=0']), # noqa
+    'O2_closure': (False, False, ['-O2', '--closure', '1']), # noqa
+    'O2_closure_emit': (True, False, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE', '--closure', '1']), # noqa
+    'O2_closure_js_emit': (True, False, ['-O2', '-s', 'EMIT_EMSCRIPTEN_LICENSE', '--closure', '1', '-s', 'WASM=0']), # noqa
   })
-  def test_emscripten_license(self, expect_license, args):
+  def test_emscripten_license(self, expect_license, allow_multiple, args):
     # fastcomp does not support the new license flag
     if not self.is_wasm_backend():
       expect_license = False
     run_process([PYTHON, EMCC, path_from_root('tests', 'hello_world.c')] + args)
     with open('a.out.js') as f:
       js = f.read()
-    self.assertContainedIf('Copyright 2010 Emscripten authors', js, expect_license)
-    self.assertContainedIf('SPDX-License-Identifier: MIT', js, expect_license)
-    if expect_license:
-      self.assertEqual(js.count('Copyright 2010 Emscripten authors'), 1)
+    licenses_found = len(re.findall('Copyright [0-9]* The Emscripten Authors', js))
+    if expect_license and licenses_found == 0:
+      self.fail('Unable to find license block in output file!')
```

Comment:
Probably better written as:

```
if expect_license:
    self.assertEqual(licenses_found, 0, 'Unable to find license block in output file!')
```

Any time you see `self.fail` in normal test code its probably a sign something could be improved (IMHO).